src/main/java/games/strategy/engine/framework/startup/login/ClientLogin.java

```diff
@@ -7,36 +7,69 @@
 import javax.swing.JOptionPane;
 import javax.swing.JPasswordField;
 
+import com.google.common.annotations.VisibleForTesting;
+
 import games.strategy.engine.ClientContext;
 import games.strategy.net.IConnectionLogin;
 import games.strategy.util.CountDownLatchHandler;
 import games.strategy.util.EventThreadJOptionPane;
-import games.strategy.util.MD5Crypt;
 
+/**
+ * The client side of the peer-to-peer network game authentication protocol.
+ *
+ * <p>
+ * In the peer-to-peer network game authentication protocol, the client receives a challenge from the server. The client
+ * is responsible for obtaining the game password from the user and using it to send a response to the server's
+ * challenge proving that the client knows the game password.
+ * </p>
+ */
 public class ClientLogin implements IConnectionLogin {
   public static final String ENGINE_VERSION_PROPERTY = "Engine.Version";
-  public static final String JDK_VERSION_PROPERTY = "JDK.Version";
-  public static final String PASSWORD_PROPERTY = "Password";
+  private static final String JDK_VERSION_PROPERTY = "JDK.Version";
```

Comment:
Confirmed that Checkstyle is not reporting a violation on this line.  However, that could be because we don't have the ConstantName rule enabled in our ruleset.

But even if we do enable ConstantName, why are you expecting this name to trigger a violation?  The default pattern for that rule is `^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$`, which is just the historical Java naming convention for `static` `final` fields.